### ISSUE TITLE
Newest Version - no output, no logs, no clue

### ISSUE THREAD

**Issue by otakukunsan** [hilts-vaughan/UnlinkMKV-GUI#27]
Hi,

It took me hours to get your program to run under Windows 7, so please, forgive me if I come across seeming really stupid right now.

I honestly have no idea why Log::Log4Perl appears to be a critical module for unlinking previously linked chapters, but I guess it is. I installed Strawberry Perl, verified that XML::LogXML (?) was installed correctly, but Log4Perl failed a test and refused to install. I tried a forced install, no test install, and a forced with no test install, and they all failed because a test was still run, and the test failed. I kept scrolling back through the CLI, and Log::Dispatch kept appearing, so I decided that I either didn't understand the output, or the output was just vague garbage. After going through a bunch of useless information in forums, I decided to install Log::Dispatch before installing Log::Log4Perl, because the only test I shouldn't be able to disable is a test for prerequisite dependencies. It worked, but I was ready to call it quits by then.

Just for the sake of trying I tossed everything into the same directory and manually edited the ini file. I had zero expectations at this point for a number of reasons. I know that some openings and endings have karaoke added through After Effects, and things like bit depth, frame rate, CRF and bit rate, and even screen resolution don't merge or encode well when combined with the "main" video. I was happily surprised to find that my test video, episode 91 of the 1080p version of Yu Yu Hakusho seemed to be successful. Maybe I'm wrong, but if the FFMPEG that's included is really from 2015, it could cause corruption when encoding to unlink ordered chapters if it's too out of date, or incompatible with playing back x265.

Anyway, I threw your GUI into the same directory and tried to unlink the next episode, and... nothing. Seriously, nothing. The video showed up in the GUI, and it claimed the unlinking was successful, but there was nothing in the output folder. Verbose didn't provide any additional information, but I suspect a path, as in a static or dynamic link could be my problem. I manually edited the ini because the input sections in the GUI don't change background color, and I guess because of my desktop theme the inputs look grayed out to me. I've seen you mention "Native Mode" several times, but all I see in the GUI is "Native Backend," which I'm guessing is what you meant. I tried it and it failed badly because I didn't have MediaInfo in the same directory, but I'm not sure it would have worked then, too.

I ran the GUI several times only to see it fail, and that was before I fixed the Log4Perl dependency issue on my machine. The GUI ran through 11 episodes like a second or two per episode, and it did the same thing even after I got Log4Perl installed. Then after actually unlinking one episode using UnlinkMKV through the CLI, there's simply no way that my machine can extract, append subs and then merge the opening, ending and main video in two seconds. I didn't know if the CLI would appear on the desktop, minimized, or be running silent, so I can only guess that the GUI isn't properly calling to the executable. They're both in the same directory, as are my linked video files, so I'm not sure why the executable works but the GUI doesn't. I've searched the usual suspects for any log files related to the executable or GUI, Temp folders, App Data, Programs Folder and Data, Event Viewer, Windows, etc., and no logs there or in the same folder as the executable or GUI. Honestly, I've got no idea what you mean by, "If the issue is part of the UnlinkMKV core, it will be addressed there."

I'm not complaining, but if you have any thoughts on what I'm doing wrong I could definitely use the input.

Thanks.

**Comment by hilts-vaughan** [comment]
Hi there,

I'm sorry you went through that. Please understand I provide this software as-is because it was useful to me. Let me see if I can find your issue and give you some help.

**Comment by hilts-vaughan** [comment]
It seems like it's at least working for you, since you managed to unlink at least one of the episodes. Can you tried any other file? Is this file a problem?

It's a shame that Strawberry Perl has decided to act up. There is no need for Perl if you use native mode -- but as you've noted, you require mediainfo for this mode to work. I admit, this isn't the most user friendly front-end but was designed to be "good enough" to get the job done.

You can find my e-mail on my Github profile -- feel free to connect with me or upload the files somewhere and I can give you some help.

**Comment by otakukunsan** [comment]
No, it's fine. The GUI would be easier for me, and I appreciate you building one. It's only fair that I do as much as I can on my own before asking for someone else's help. At this point I can't locate any log files, so maybe the GUI isn't working because I'm missing a module for Perl. Honestly, I just hit a wall so I thought now would be the right time to ask for help, that's all.

**Comment by otakukunsan** [comment]
The GUI does not work, but it says that it is working.

I can use the executable using the command prompt, CLI, but the GUI seems to be just stepping through recursion or a loop.

**Comment by hilts-vaughan** [comment]
When it says it's working, it's often because the Perl script exited prematurely but did not provide any meaningful output. If you can run the script manually through Perl, it will probably given you the reasons. I think if you tick off "Verbose" mode it's supposed to show the commands as they execute. Does it still do this or no?

Vaughan

**Comment by otakukunsan** [comment]
First, Perl.exe does run when I click "Unlink" in your GUI, and it also displays in TaskManager along with your GUI. Something else shows up in TaskManager, but it could just be the CLI activating Perl. The list drops by one line and then goes right back up by one, so I can't scroll the list fast enough to tell what it is or what it may be related to, if anything.

The only thing that appears in your GUI as far as log information goes, is the following:

C:\Strawberry\perl\bin\perl.exe "I:\(0)\(temp)\ZZZ\UnlinkMKV-GUI.0.2.1\winport.pl"  --outdir "I:\(0)\(temp)\ZZZ\UnlinkMKV-GUI.0.2.1\1" "I:\(0)\(temp)\ZZZ\UnlinkMKV-GUI.0.2.1\UnlinkMKV-GUI.exe.config"

C:\Strawberry\perl\bin\perl.exe "I:\(0)\(temp)\ZZZ\UnlinkMKV-GUI.0.2.1\winport.pl" --ll TRACE  --outdir "I:\(0)\(temp)\ZZZ\UnlinkMKV-GUI.0.2.1\1" "I:\(0)\(temp)\ZZZ\UnlinkMKV-GUI.0.2.1\UnlinkMKV-GUI.exe.config"


I tried running the script from the command prompt in the same directory, and I think I'm still missing some dependencies, but I'm not sure. I played around a little bit a long time ago with ActiveState Perl and things like hashes and subroutines, but I never got around to learning about the modules. That's actually why it took me so long to figure out that error 255 for a failed test meant that I lacked a prerequisite dependency. lol
![verbose off](https://user-images.githubusercontent.com/30187951/28236395-7363f750-68f3-11e7-8d5f-0d323477d534.png)

**Comment by otakukunsan** [comment]
This is from running the script directly:

Can't locate String/CRC32.pm in @INC (you may need to install the String::CRC32
module) (@INC contains: C:/Strawberry/perl/site/lib C:/Strawberry/perl/vendor/li
b C:/Strawberry/perl/lib) at winport.pl line 13.
BEGIN failed--compilation aborted at winport.pl line 13.

**Comment by otakukunsan** [comment]
I installed String::CRC32, and then tried unlinking two more episodes as a test. The test failed, but this time for a different reason. Again, I'm not complaining, but I definitely do not like the reason that this test failed.

One episode was 1.1 GB, and the other 800 MB, and the test should have passed because I successfully resized and encoded these episodes, openings and endings, and that brought me to your GUI. The completed rebuilds were 162 KB and 668 MB, which literally made no sense to me. So on a hunch I went back to the ini file, and I discovered that the GUI restored the original values. I only have 3.67 GB of free space on the C: drive, which explained the strange sizes of the unlinked rebuilds. Then I manually edited the ini again, and ran another test. The GUI restores the default values to the ini file, not when you execute it, but it does so as soon as you click on the "Unlink" button, or very shortly thereafter.

Manually edited ini file:

ffmpeg = I:\(0)\(temp)\ZZZ\UnlinkMKV-GUI.0.2.1\ffmpeg\bin\ffmpeg.exe
mkvext = C:\Program Files\MKVToolNix\mkvextract.exe
mkvinfo = C:\Program Files\MKVToolNix\mkvinfo.exe
mkvmerge = C:\Program Files\MKVToolNix\mkvmerge.exe
fix_audio = 0
fix_video = 0
fix_subtitles = 1
ignoredef = 0
chapters = 1
tmpdir = I:\(temp)\UnlinkMKV
out_dir = I:\(0)\(temp)\ZZZ\UnlinkMKV-GUI.0.2.1\1


Settings of ini file while using the GUI:

ffmpeg = 
mkvext = C:\Program Files\MKVToolNix\mkvextract.exe
mkvinfo = C:\Program Files\MKVToolNix\mkvinfo.exe
mkvmerge = C:\Program Files\MKVToolNix\mkvmerge.exe
fix_audio = 0
fix_video = 0
fix_subtitles = 1
ignoredef = 0
chapters = 1
tmpdir = C:\Users\End User 0\AppData\Local\Temp\UnlinkMKV
out_dir = I:\(0)\(temp)\ZZZ\UnlinkMKV-GUI.0.2.1\1


I don't know why it's not possible to set a static path to ffmpeg, but it would really be nice to be allowed to manually define the location of the temporary directory. If at all possible, would you be willing to at least consider adding another input field into the GUI to allow for manually assigning the temp directory, or if the ini values are embedded in the script, maybe consider removing them in the near future?
